### PR TITLE
NEXUS-10329 added basic auth section

### PR DESCRIPTION
**Pull request by joedragons**
* Subsectioned the two authentication methods out
* Changed the intro slightly

JIRA: https://issues.sonatype.org/browse/NEXUS-10329
CI: http://bamboo.s/browse/BOOK-NEXUS165-5

### CODE REVIEW

File: chapter-npm.asciidoc [hunk 1]
```diff
@@ -155,6 +159,56 @@ file for the specific repository.
 
 Further details on +npm login+ can be found on the https://docs.npmjs.com/cli/adduser[npm website].
 
+==== Authentication using basic auth
```
Comment (bentmann) [inline chapter-npm.asciidoc]: Might want to double-check with @dulanism on the style for headers, from other chapters (e.g. https://github.com/sonatype/nexus-book/blob/master/chapter-eval.asciidoc#integration-with-enterprise-ldap-solutions) I get the impression they use title-case.

File: chapter-npm.asciidoc [hunk 2]
```diff
@@ -155,6 +159,56 @@ file for the specific repository.
...
+
+The +_auth+ variable has to be generated by base64-encoding the string of +username:password+.  You can create 
+this encoded string with the command line call to +openssl+ e.g. for the default +admin+ user:
+
+----
+echo -n 'admin:admin123' | openssl base64
+----
+
+Other tools for the encoding are +uuencode+ or, for Windows users, +certutil+. To use +certutil+ on Windows you 
+need to put the credentials to be encoded into a file:
```
Comment (bentmann) [inline chapter-npm.asciidoc]: We might want to call out that that file shouldn't contain additional whitespace or a trailing line separator

File: chapter-npm.asciidoc [hunk 3]
```diff
@@ -155,6 +159,56 @@ file for the specific repository.
...
+-----BEGIN CERTIFICATE-----
+YWRtaW46YWRtaW4xMjM=
+-----END CERTIFICATE-----
+----
+
+TIP: Whatever tool you use to generate the encoded username and password string can be tested by encoding the 
+string +admin:admin123+, which should result in +YWRtaW46YWRtaW4xMjM=+. Another example is 
++jane:testpassword123+ which should result in +amFuZTp0ZXN0cGFzc3dvcmQxMjM=+.
+
+Once you have the encoded credentials, it as well as author email and enabling of authentication can then be
```
Comment (bentmann) [inline chapter-npm.asciidoc]: "**the** author email"?

**Comment by bentmann** [comment]
+1

**Comment by joedragons** [comment]
Thanks @bentmann .  I thought of another thing this AM I want to check (tangential), so moved back to WIP.

File: chapter-npm.asciidoc (hunk 2 above)
Comment (joedragons) [inline chapter-npm.asciidoc]: I copied this from the NX2 guide, I will look into bettering.

File: chapter-npm.asciidoc (hunk 1 above)
Comment (dulanism) [inline chapter-npm.asciidoc]: Using the precedent of other chapters across the book, title-case should be the case for "Authentication using basic auth". It should thus be "Authentication Using Basic Auth".

File: chapter-npm.asciidoc (hunk 2 above)
Comment (joedragons) [inline chapter-npm.asciidoc]: Added TIP:
See 5520125.

File: chapter-npm.asciidoc [hunk 4]
```diff
@@ -155,6 +159,59 @@ file for the specific repository.
 
 Further details on +npm login+ can be found on the https://docs.npmjs.com/cli/adduser[npm website].
 
+==== Authentication Using Basic Auth
+
+If you cannot use the realm and login method, for example if you have a username which includes capital letters
+(disallowed by npm login), you can configure npm to use basic auth with your repository manager. This 
+authentication method involves editing the +.npmrc+ configuration file adding an encoded username and password as
```
Comment (dulanism) [inline chapter-npm.asciidoc]: Minor: Can you simplify the sentences to make two? 
Original - If you cannot use the realm and login method, for example if you have a username which includes capital letters (disallowed by npm login), you can configure npm to use basic auth with your repository manager.

Improvement - In some instances you cannot use the realm and login method. For example, if you have a username which includes capital letters (disallowed by `npm login`), you can configure npm to use basic auth with your repository manager. 

(Footnote - I added the "`" markup to npm login, as it's marked up in the "Once the realm..." sentence in the section above)

File: chapter-npm.asciidoc [hunk 5]
```diff
@@ -155,6 +159,59 @@ file for the specific repository.
...
+----
+
+Other tools for the encoding are +uuencode+ or, for Windows users, +certutil+. To use +certutil+ on Windows you 
+need to put the credentials to be encoded into a file:
+
+----
+admin:admin123
+----
+
+TIP: Assure your file does not have extra whitespace or a trailing line seperator as either of these will
```
Comment (dulanism) [inline chapter-npm.asciidoc]: Minor: Ensure and Assure are definitely cousins in definition. But I wanted to check my AP style guide just to be sure ... 
> To “assure” a person of something is to make him or her confident of it. To “ensure” that something happens is to make certain that it does.

I think the choice of words is matter of someone vs something, imo

File: chapter-npm.asciidoc [hunk 6]
```diff
@@ -155,6 +159,59 @@ file for the specific repository.
...
+-----BEGIN CERTIFICATE-----
+YWRtaW46YWRtaW4xMjM=
+-----END CERTIFICATE-----
+----
+
+TIP: Whatever tool you use to generate the encoded username and password string can be tested by encoding the 
+string +admin:admin123+, which should result in +YWRtaW46YWRtaW4xMjM=+. Another example is 
++jane:testpassword123+ which should result in +amFuZTp0ZXN0cGFzc3dvcmQxMjM=+.
+
+Once you have the encoded credentials, it as well as the author email and enabling of authentication can then be
```
Comment (dulanism) [inline chapter-npm.asciidoc]: "Once you have the encoded credentials, it as well as the author email and enabling of authentication can then be then be added to the `.npmrc` file (below your already entered registry config)."

Maybe something like -
Once your credentials are encoded they can be added to the `.npmrc` file, along with your author email and enabled authentication (below your already entered registry configuration). 

Sorry if splitting hairs

**Comment by dulanism** [comment]
+1 (minor sentence structure/grammar suggestions)

File: chapter-npm.asciidoc (hunk 4 above)
Comment (joedragons) [inline chapter-npm.asciidoc]: @dulanism How do you feel about:
"In some instances you cannot use the realm and login method, for example if you have a username which includes capital letters (disallowed by +npm login+).  In these you can still use npm by configuring it to use basic auth with your repository manager."
The example seemed weird to me tied to the instructional sentence.  If you prefer yours, I will defer.

File: chapter-npm.asciidoc (hunk 5 above)
Comment (joedragons) [inline chapter-npm.asciidoc]: Addressed in next commit, thanks!

File: chapter-npm.asciidoc (hunk 6 above)
Comment (joedragons) [inline chapter-npm.asciidoc]: I'm neutral so I made the change=)  Thanks.

File: chapter-npm.asciidoc (hunk 4 above)
Comment (dulanism) [inline chapter-npm.asciidoc]: LGTM @joedragons

**Comment by joedragons** [comment]
Squash merged: d2db1ff

**Pull request closed without merging**